src/suites/cts/validation/createBindGroup.spec.ts

```diff
@@ -66,49 +66,34 @@ g.test('binding must be present in layout', async t => {
 });
 
 g.test('buffer binding must contain exactly one buffer of its type', t => {
-  const { bindingType, resourceType, resourceIsError } = t.params;
+  const bindingType: GPUBindingType = t.params.bindingType;
+  const resourceType: BindingResourceType = t.params.resourceType;
 
   const layout = t.device.createBindGroupLayout({
     bindings: [{ binding: 0, visibility: GPUShaderStage.COMPUTE, type: bindingType }],
   });
 
-  const resource = t.getBindingResource(resourceType, resourceIsError);
-
-  let shouldError = bindingType !== resourceType;
-  if (bindingType === 'readonly-storage-buffer' && resourceType === 'storage-buffer') {
-    shouldError = false;
-  }
-  if (resourceIsError) {
-    shouldError = true;
-  }
+  const resource = t.getBindingResource(resourceType);
 
+  const shouldError = !resourceBindingMatches(bindingType, resourceType);
```

Comment:
won't this hit the `unreachable`?

Reply:
which unreachable and in what situation? It doesn't seem to at run time.